src/config/source.rs

```diff
@@ -140,7 +140,8 @@ impl SourceContext {
         }
     }
 
-    pub fn do_acknowledgements(&self, config: &AcknowledgementsConfig) -> bool {
+    pub fn do_acknowledgements<C: Into<AcknowledgementsConfig>>(&self, config: C) -> bool {
```

Comment:
Isn't `C` always going to be `SourceAcknowledgementsConfig` here? Did I miss a call site?

Reply:
Hmm, yeah, that's a fair point. I know I did this originally because it let me change less code... but you're right, I believe: it doesn't actually have to be generic.

I'll clean this up.

Reply:
Ended up having to add `into` in a few places, so I was able to clean this up a little, which does keep it a bit more straightforward.